packages/twenty-front/src/modules/object-record/record-index/hooks/useHandleIndexIdentifierClick.ts

```diff
@@ -0,0 +1,55 @@
+import { ObjectMetadataItem } from '@/object-metadata/types/ObjectMetadataItem';
+import { useFindRecordCursorFromFindManyCacheRootQuery } from '@/object-record/record-show/hooks/useFindRecordCursorFromFindManyCacheRootQuery';
+import { buildShowPageURL } from '@/object-record/record-show/utils/buildShowPageURL';
+import { useQueryVariablesFromActiveFieldsOfViewOrDefaultView } from '@/views/hooks/useQueryVariablesFromActiveFieldsOfViewOrDefaultView';
+import { currentViewIdComponentState } from '@/views/states/currentViewIdComponentState';
+import { useNavigate } from 'react-router-dom';
+import { useRecoilValue } from 'recoil';
+
+export const useHandleIndexIdentifierClick = ({
+  objectMetadataItem,
+  recordIndexId,
+}: {
+  recordIndexId: string;
+  objectMetadataItem: ObjectMetadataItem;
+}) => {
+  const navigate = useNavigate();
+
+  const currentViewId = useRecoilValue(
+    currentViewIdComponentState({
+      scopeId: recordIndexId,
+    }),
+  );
+
+  const { filter, orderBy } =
+    useQueryVariablesFromActiveFieldsOfViewOrDefaultView({
+      objectMetadataItem,
+      viewId: currentViewId,
+    });
+
+  const { findCursorInCache } = useFindRecordCursorFromFindManyCacheRootQuery({
```

Comment:
is this right? what we we have several findMany on the same filters and orderBy?